stackslib/src/chainstate/burn/db/sortdb.rs

```diff
@@ -2848,6 +2848,35 @@ impl SortitionDB {
         Ok(())
     }
 
+    /// Validates given StacksEpochs (will runtime panic if there is any invalid StacksEpoch structuring) and
+    /// replaces them into the SortitionDB's epochs table
+    fn validate_and_replace_epochs(
+        db_tx: &Transaction,
+        epochs: &[StacksEpoch],
+    ) -> Result<(), db_error> {
+        let epochs = StacksEpoch::validate_epochs(epochs);
+        let existing_epochs = Self::get_stacks_epochs(db_tx)?;
+        if existing_epochs == epochs {
+            return Ok(());
+        }
```

Comment:
I'd prefer to see some kind of early erroring here in the event that `diff(existing_epochs, epochs).start_height` has been reached already by the sortition db.

Reply:
Sure, one sec